query/graphql/src/main/java/org/codice/ddf/admin/graphql/transform/GraphQLTransformCommons.java

```diff
@@ -38,6 +46,36 @@ public GraphQLTransformCommons() {
         return transformOutput.fieldsToGraphQLFieldDefinition(Arrays.asList(provider));
     }
 
+    public GraphQLQueryProvider getErrorCodesQueryProvider(List<FieldProvider> fieldProviders) {
+        Set<String> errorCodes = new TreeSet<>();
+
+        for (FieldProvider fieldProvider : fieldProviders) {
+            List<FunctionField> mutations = fieldProvider.getMutationFunctions();
+            List<Field> queryFields = fieldProvider.getDiscoveryFields();
+
+            for (FunctionField mutation : mutations) {
+                errorCodes.addAll(mutation.getErrorCodes());
+            }
+
+            for (Field field : queryFields) {
+                errorCodes.addAll(field.getErrorCodes());
+            }
+        }
+
+        GraphQLEnumType.Builder enumTypeBuilder = GraphQLEnumType.newEnum()
+                .name("ErrorCode")
+                .description("All possible error codes.");
+        errorCodes.forEach(enumTypeBuilder::value);
+        GraphQLEnumType errorCodeEnumType = enumTypeBuilder.build();
+
+        return () -> Collections.singletonList(GraphQLFieldDefinition.newFieldDefinition()
```

Comment:
❓ Should we be creating this through our internal API as its own function called `errorCodes`? Doesn't seem consistent with how we are doing everything else.